src/pam_hbac_utils.c

```diff
@@ -83,12 +83,31 @@ null_cstring_array_size(const char *arr[])
     return nelem;
 }
 
+static bool debug_mode = true;
+
+void set_debug_mode(bool v)
+{
+    debug_mode = v;
+}
+
 void logger(pam_handle_t *pamh, int level, const char *fmt, ...)
 {
     va_list ap;
 
     va_start(ap, fmt);
 
+    logger2(pamh, level, fmt, ap);
+
+    va_end(ap);
+}
+
+void logger2(pam_handle_t *pamh, int level, const char *fmt, va_list ap)
+{
+    /* If not in debug mode then do not be verbose. */
+    if (debug_mode == false && level > LOG_ERR) {
```

Comment:
LOG_ERR is not displayed? It should be! I thought I checked that...I'll recheck tomorrow.